### ISSUE TITLE
v8.5.4: update collection repeater field options get Directus\Exception\SQLException

### ISSUE THREAD

**Issue by everyx** [directus/api#1749]
• Version of Directus: v8.5.4
• Error message (if any)
```shell
api_1             | [2020-02-08 12:27:18] api[xxxxxx].ERROR: Directus\Exception\SQLException: Statement could not be executed (42000 - 1064 - You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '(16777215) Null' at line 1) in /var/www/html/src/endpoints/Fields.php:122
api_1             | Stack trace:
api_1             | #0 [internal function]: Directus\Api\Routes\Fields->update
api_1             | #1 /var/www/html/vendor/slim/slim/Slim/Handlers/Strategies/RequestResponse.php(40): call_user_func
api_1             | #2 /var/www/html/vendor/slim/slim/Slim/Route.php(281): Slim\Handlers\Strategies\RequestResponse->__invoke
api_1             | #3 /var/www/html/src/core/Directus/Application/Http/Middleware/AbstractRateLimitMiddleware.php(34): Slim\Route->__invoke
api_1             | #4 [internal function]: Directus\Application\Http\Middleware\AbstractRateLimitMiddleware->__invoke
api_1             | #5 /var/www/html/vendor/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array
api_1             | #6 [internal function]: Slim\DeferredCallable->__invoke
api_1             | #7 /var/www/html/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func
api_1             | #8 /var/www/html/src/core/Directus/Application/Http/Middleware/AuthenticationMiddleware.php(124): Slim\Route->Slim\{closure}
api_1             | #9 [internal function]: Directus\Application\Http\Middleware\AuthenticationMiddleware->__invoke
api_1             | #10 /var/www/html/vendor/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array
api_1             | #11 [internal function]: Slim\DeferredCallable->__invoke
api_1             | #12 /var/www/html/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func
api_1             | #13 /var/www/html/src/core/Directus/Application/Http/Middleware/TableGatewayMiddleware.php(25): Slim\Route->Slim\{closure}
api_1             | #14 [internal function]: Directus\Application\Http\Middleware\TableGatewayMiddleware->__invoke
api_1             | #15 /var/www/html/vendor/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array
api_1             | #16 [internal function]: Slim\DeferredCallable->__invoke
api_1             | #17 /var/www/html/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func
api_1             | #18 /var/www/html/src/core/Directus/Application/Http/Middleware/DatabaseMigrationMiddleware.php(15): Slim\Route->Slim\{closure}
api_1             | #19 [internal function]: Directus\Application\Http\Middleware\DatabaseMigrationMiddleware->__invoke
api_1             | #20 /var/www/html/vendor/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array
api_1             | #21 [internal function]: Slim\DeferredCallable->__invoke
api_1             | #22 /var/www/html/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func
api_1             | #23 /var/www/html/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(117): Slim\Route->Slim\{closure}
api_1             | #24 /var/www/html/vendor/slim/slim/Slim/Route.php(268): Slim\Route->callMiddlewareStack
api_1             | #25 /var/www/html/vendor/slim/slim/Slim/App.php(503): Slim\Route->run
api_1             | #26 /var/www/html/src/core/Directus/Application/Http/Middleware/AbstractRateLimitMiddleware.php(34): Slim\App->__invoke
api_1             | #27 [internal function]: Directus\Application\Http\Middleware\AbstractRateLimitMiddleware->__invoke
api_1             | #28 /var/www/html/vendor/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array
api_1             | #29 [internal function]: Slim\DeferredCallable->__invoke
api_1             | #30 /var/www/html/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func
api_1             | #31 /var/www/html/vendor/directus/proxy-detection/src/ProxyDetectionMiddleware.php(30): Slim\App->Slim\{closure}
api_1             | #32 /var/www/html/src/core/Directus/Application/Http/Middleware/ProxyMiddleware.php(18): RKA\Middleware\ProxyDetectionMiddleware->__invoke
api_1             | #33 [internal function]: Directus\Application\Http\Middleware\ProxyMiddleware->__invoke
api_1             | #34 /var/www/html/vendor/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array
api_1             | #35 [internal function]: Slim\DeferredCallable->__invoke
api_1             | #36 /var/www/html/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func
api_1             | #37 /var/www/html/vendor/akrabat/ip-address-middleware/src/IpAddress.php(113): Slim\App->Slim\{closure}
api_1             | #38 [internal function]: RKA\Middleware\IpAddress->__invoke
api_1             | #39 /var/www/html/vendor/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array
api_1             | #40 [internal function]: Slim\DeferredCallable->__invoke
api_1             | #41 /var/www/html/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func
api_1             | #42 /var/www/html/src/core/Directus/Application/Http/Middleware/CorsMiddleware.php(71): Slim\App->Slim\{closure}
api_1             | #43 [internal function]: Directus\Application\Http\Middleware\CorsMiddleware->__invoke
api_1             | #44 /var/www/html/vendor/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array
api_1             | #45 [internal function]: Slim\DeferredCallable->__invoke
api_1             | #46 /var/www/html/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func
api_1             | #47 /var/www/html/src/core/Directus/Application/Http/Middleware/ResponseCacheMiddleware.php(62): Slim\App->Slim\{closure}
api_1             | #48 [internal function]: Directus\Application\Http\Middleware\ResponseCacheMiddleware->__invoke
api_1             | #49 /var/www/html/vendor/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array
api_1             | #50 [internal function]: Slim\DeferredCallable->__invoke
api_1             | #51 /var/www/html/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func
api_1             | #52 /var/www/html/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(117): Slim\App->Slim\{closure}
api_1             | #53 /var/www/html/vendor/slim/slim/Slim/App.php(392): Slim\App->callMiddlewareStack
api_1             | #54 /var/www/html/vendor/slim/slim/Slim/App.php(297): Slim\App->process
api_1             | #55 /var/www/html/src/core/Directus/Application/Application.php(161): Slim\App->run
api_1             | #56 /var/www/html/public/index.php(5): Directus\Application\Application->run [] []
```
• Steps to reproduce
This is a reproduce in Directu Demo site.
![bug](https://user-images.githubusercontent.com/1665522/74085310-d831d500-4ab2-11ea-859c-8df650043a8c.gif)

• PHP version: 7.4.2
• MySQL version: both v5.7.28 and v8.0.18
• Webserver + version: Caddy v1.0.4

Maybe is the same problem of https://github.com/directus/api/issues/1722

**Comment by everyx** [comment]
I have patched #1727, looks like it could fix this issue.

**Comment by everyx** [comment]
Also reproduced in v8.5.5 at directus demo site

**Comment by hemratna** [comment]
@everyx 
I fixed this issue with PR #1764.
Please check it and if issue is solved then close it.

**Comment by rijkvanzanten** [comment]
Fixed in https://github.com/directus/api/pull/1764
